app/models/user.rb

```diff
@@ -1034,10 +1034,9 @@ def email_confirmed?
     single_sign_on_record&.external_email&.downcase == email
   end
 
-  def activate
-    if email_token = self.email_tokens.active.where(email: self.email).first
-      EmailToken.confirm(email_token.token, skip_reviewable: true)
-    end
+  def activate # TODO: is this duplicated?
```

Comment:
Is it?

Reply:
Yes, this method and `UserAuthenticator#confirm_email` do roughly the same thing in a different way. It's fixed now. :)